File: src/main/org/nlogo/prim/threed/_rollright.java
package org.nlogo.prim.threed;

import org.nlogo.agent.Turtle3D;
import org.nlogo.api.LogoException;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Syntax;

public final strictfp class _rollright
	extends Command
{
	@Override
	public Syntax syntax()
	{
		return Syntax.commandSyntax
			( new int[] { Syntax.TYPE_NUMBER } ,
			  "-T--" , true ) ;
	}
	@Override
	public void perform( final Context context )
		throws LogoException
	{
		double delta = argEvalDoubleValue( context , 0 ) ;
		Turtle3D turtle = (Turtle3D) context.agent ;
		turtle.roll( turtle.roll() + delta ) ;
		context.ip = next ;
	}
}

File: netlogo-gui/src/main/prim/threed/_rollright.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.threed

import org.nlogo.agent.Turtle3D
import org.nlogo.nvm.{ Command, Context }

class _rollright extends Command {


  switches = true
  override def perform(context: Context): Unit = {
    val delta = argEvalDoubleValue(context, 0)
    val t = context.agent.asInstanceOf[Turtle3D]
    t.roll(t.roll + delta)
    context.ip = next
  }
}
